kaiax/gov/headergov/impl/init.go

```diff
@@ -259,3 +257,71 @@ func validateOpts(opts *InitOpts) error {
 
 	return nil
 }
+
+// GetGenesisGovBytes returns the genesis governance bytes for initGenesis().
+// It panics if the given chain config is invalid.
+func GetGenesisGovBytes(config *params.ChainConfig) headergov.GovBytes {
+	partialParamSet := make(gov.PartialParamSet)
+	genesisParamNames := getGenesisParamNames(config)
+	for _, name := range genesisParamNames {
+		param := gov.Params[name]
+		value, err := param.ChainConfigValue(config)
+		if err != nil {
+			panic(err)
+		}
+
+		err = partialParamSet.Add(string(name), value)
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	govData := headergov.NewGovData(partialParamSet)
+	ret, err := govData.ToGovBytes()
+	if err != nil {
+		panic(err)
+	}
+
+	return ret
+}
+
+func getGenesisParamNames(config *params.ChainConfig) []gov.ParamName {
+	genesisParamNames := make([]gov.ParamName, 0)
+
```

Comment:
- Please comment something like "Note that DeriveShaImpl is not included in genesisBlock.Governance for backwards compatibility"
- In old code https://github.com/klaytn/klaytn/blob/848b00c4a05222c4f74ade60e9ddbe0b9595db2a/blockchain/genesis.go#L464 the UnitPrice field was not in `config.Genesis != nil` condition. The behavior was added later https://github.com/klaytn/klaytn/blob/848b00c4a05222c4f74ade60e9ddbe0b9595db2a/governance/default.go#L1090 and the behavior seems a peculiar bug?

Reply:
I didn't fully get what you're saying, but `genesis.Governance` in DB is created differently:
- Private net, `ken init`: `GetGovernanceItemsFromChainConfig` is used, and `SetGenesisGovernance` is not called. It hits this line followed by `genesis.Commit`:
https://github.com/kaiachain/kaia/blob/0412649e00e73edb757ad706a8d93345d6b737b5/blockchain/genesis.go#L209

- Mainnet/Kairos, the very first launch without init: `SetGenesisGovernance` is used, followed by `genesis.Commit`:
https://github.com/kaiachain/kaia/blob/0412649e00e73edb757ad706a8d93345d6b737b5/blockchain/genesis.go#L206

- All networks, launch after init: `SetGenesisGovernance` is called, but *not* committed to DB. Thus, `genesis.Governance` in memory contains the initial thirteen parameters regardless of the genesis in DB, but seems not to be used.
https://github.com/kaiachain/kaia/blob/0412649e00e73edb757ad706a8d93345d6b737b5/blockchain/genesis.go#L242

Reply:
It is very unlikely corner case where `genesis.Config.Governance = nil`
- In SetGenesisGovernance, result includes governance.unitprice
- In GetGovernanceItemsFromChainConfig and getGenesisParamNames, result does not include governance.unitprice.
But no use case has `genesis.Config.Governance = nil` though.

Reply:
If `genesis.Config.Governance = nil`, then `SetGenesisGovernance` is never called. Even if it is called, it's going to cause nil dereference panic.

Reply:
Oh, you're right.